### PR TITLE
JAMES-3737 More control on configuring threads

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

File: protocols/netty/src/main/java/org/apache/james/protocols/netty/AbstractAsyncServer.java [hunk 1]
```diff
@@ -102,10 +123,12 @@ public synchronized void bind() throws Exception {
         ServerBootstrap bootstrap = new ServerBootstrap();
         bootstrap.channel(NioServerSocketChannel.class);
 
-        bossGroup = new NioEventLoopGroup(2, NamedThreadFactory.withName(jmxName + "-boss"));
+        bossGroup = bossWorker.map(count -> new NioEventLoopGroup(2, NamedThreadFactory.withName(jmxName + "-boss")));
```
Comment (quantranhong1999) [inline protocols/netty/src/main/java/org/apache/james/protocols/netty/AbstractAsyncServer.java:126]:
```suggestion
        bossGroup = bossWorker.map(count -> new NioEventLoopGroup(count, NamedThreadFactory.withName(jmxName + "-boss")));
```

File: protocols/netty/src/main/java/org/apache/james/protocols/netty/AbstractAsyncServer.java [hunk 2]
```diff
@@ -49,19 +50,21 @@
...
+    private Optional<EventLoopGroup> bossGroup;
     private EventLoopGroup workerGroup;
 
     private volatile boolean started;
     
     private final ChannelGroup channels = new DefaultChannelGroup(ImmediateEventExecutor.INSTANCE);
 
     private volatile int ioWorker = DEFAULT_IO_WORKER_COUNT;
-    
+    private volatile Optional<Integer> bossWorker = Optional.of(DEFAULT_IO_WORKER_COUNT);
```
Comment (vttranlina) [inline protocols/netty/src/main/java/org/apache/james/protocols/netty/AbstractAsyncServer.java:66]:
```suggestion
    private volatile Optional<Integer> bossWorker = Optional.of(DEFAULT_BOSS_WORKER_COUNT);
```

File: protocols/netty/src/main/java/org/apache/james/protocols/netty/AbstractAsyncServer.java (hunk 1 above)
Comment (vttranlina) [inline protocols/netty/src/main/java/org/apache/james/protocols/netty/AbstractAsyncServer.java:126]:
```suggestion
        bossGroup = bossWorker.map(count -> new NioEventLoopGroup(count, NamedThreadFactory.withName(jmxName + "-boss")));
```

Comment (quantranhong1999) [reply protocols/netty/src/main/java/org/apache/james/protocols/netty/AbstractAsyncServer.java:126]: You almost spot the same as me :p

**Review by quantranhong1999 — APPROVED** [review]
```
org.apache.james.mpt.imapmailbox.rabbitmq.RabbitMQConcurrentSessionsTest.testConcurrentFetchResponseUS

Error Message
Location: /org/apache/james/imap/scripts/ConcurrentFetchResponse.test:114
LastClientMsg: 3b COPY 2 inbox
Expected: '\* 1 FETCH \(FLAGS \(\\Flagged\)\)'
Actual   : '3b NO COPY processing failed.'
```

**Pull request merged**